var/spack/repos/builtin/packages/ascent/package.py

```diff
@@ -185,7 +185,11 @@ def install(self, spec, prefix):
         with working_dir('spack-build', create=True):
             py_site_pkgs_dir = None
             if "+python" in spec:
-                py_site_pkgs_dir = site_packages_dir
+                try:
+                    py_site_pkgs_dir = site_packages_dir
+                except NameError:
+                    # spack's  won't exist in a subclass
```

Comment:
I don't understand the meaning of this comment. Can you expand on it? I was wondering why:
```python
py_site_pkgs_dir = site_packages_dir if self.spec.satisfies('+python') else None
```
was not sufficient and we have to use both a conditional and a `try/except` clause.

Reply:
@alalazo b/c of how spack's classes and extensions are implemented `site_packages_dir` won't exist when a class inherits from this class. `site_packages_dir` is a bit magic.  The comment could be improved, but we do inherit fro this class in our development workflows.

Reply:
Ok, I get this now - in fact `site_packages_dir` is monkey patched by Python at a module level so I think you would get it in the module where the subclass is defined. We need to fix that behavior. @adamjstewart fyi